pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -549,18 +553,66 @@ func (os *OpenStack) GetZone() (cloudprovider.Zone, error) {
 	return zone, nil
 }
 
+func (os *OpenStack) getZoneByServer(server *servers.Server) (cloudprovider.Zone, error) {
+	az := ""
+	var ok bool
+
+	if az, ok = server.Metadata[AvailabilityZone]; !ok {
+		return cloudprovider.Zone{}, fmt.Errorf("Can not get %s from the Metadata of instance %s", AvailabilityZone, server.Name)
+	}
+	if len(az) == 0 {
+		return cloudprovider.Zone{}, fmt.Errorf("The %s of instance %s is empty.", AvailabilityZone, server.Name)
+	}
+
+	zone := cloudprovider.Zone{
+		FailureDomain: az,
+		Region:        os.region,
+	}
+	glog.V(1).Infof("Current zone is %v", zone)
```

Comment:
Does this need to be `V(1)`? Can we make it `V(4)`

Reply:
Right, I update it soon.